RetailCoder.VBE/Refactorings/Rename/RenameRefactoring.cs

```diff
@@ -10,357 +9,558 @@
+            RequestParseAfterRename = false;
 
-            if (declarations.Any())
-            {
-                var message = string.Format(RubberduckUI.RenameDialog_ConflictingNames, _model.NewName,
-                    declarations.FirstOrDefault().IdentifierName);
+            RenameReferences(_model.Target, _model.NewName);
 
-                var rename = _messageBox.Show(message, RubberduckUI.RenameDialog_Caption, MessageBoxButtons.YesNo,
-                    MessageBoxIcon.Exclamation);
+            if (_model.Target.DeclarationType.HasFlag(DeclarationType.ClassModule))
+            {
+                foreach (var reference in _model.Target.References)
+                {
+                    var ctxt = ParserRuleContextHelper.GetParent<VBAParser.ImplementsStmtContext>(reference.Context);
+                    if (ctxt != null)
+                    {
+                        RenameDefinedFormatMembers(_state.DeclarationFinder.FindInterfaceMembersForImplementsContext(ctxt), _appendUnderscoreFormat);
+                    }
+                }
+            }
 
-                _renameOperationIsCancelled = (rename == DialogResult.No);
+            var component = _model.Target.QualifiedName.QualifiedModuleName.Component;
+            if (component.Type == ComponentType.Document)
+            {
+                var properties = component.Properties;
+                var property = properties["_CodeName"];
+                {
+                    property.Value = _model.NewName;
+                }
+            }
+            else if (component.Type == ComponentType.UserForm)
+            {
+                var properties = component.Properties;
+                var property = properties["Caption"];
+                {
+                    if ((string)property.Value == _model.Target.IdentifierName)
+                    {
+                        property.Value = _model.NewName;
+                    }
+                    component.Name = _model.NewName;
+                }
+            }
+            else
+            {
+                Debug.Assert(!component.CodeModule.IsWrappingNullReference, "input validation fail: Attempting to rename an ICodeModule wrapping a null reference");
+                component.CodeModule.Name = _model.NewName;
             }
-            return _renameOperationIsCancelled;
         }
 
-        private bool IsControlEventHandler(Declaration userTarget, out Declaration control)
+        private void RenameProject()
         {
-            control = null;
-            var declarationsOfInterest = _model.State.AllUserDeclarations.Where(d => d.DeclarationType.HasFlag(DeclarationType.Control)
-                    && userTarget.IdentifierName.StartsWith(d.IdentifierName));
+            RequestParseAfterRename = false;
+            var projects = _model.VBE.VBProjects;
+            var project = projects.SingleOrDefault(p => p.HelpFile == _model.Target.ProjectId);
```

Comment:
`HelpFile` is an implementation detail of `ProjectId`; wouldn't it be better to compare `p.ProjectId` instead? That way if we change the way we ID a project then that bit won't break...